### DISCUSSION TITLE
ScriptReference/ComputeShader.SetFloats

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#690] [Notes]
# ScriptReference/ComputeShader.SetFloats

https://docs.unity3d.com/ScriptReference/ComputeShader.SetFloats.html

**Comment by qwe321qwe321qwe321** [comment]
> This API feeds raw data to the constant buffer, so the provided data must follow the HLSL constant buffer data layout rules. This means that the the array elements must be aligned on float4; for example, float4 data requires no padding, float3 data needs one float padding for each element, float2 data needs two floats, and so on.

Here is a sample for padding values in float arrays:
https://cmwdexint.com/2017/12/04/computeshader-setfloats/

C#:
```cs
 _floatArray = new float[4 * 4];
 _floatArray[0] = 1f;
 _floatArray[4] = 0.5f;
 _floatArray[8] = 0.75f;
 _floatArray[12] = 1f;
```

Compute shader:
```
float FloatArray[4];
```
